services/provider/server/server.go

```diff
@@ -189,3 +206,145 @@ func newClient() (client.Client, error) {
+	if err != nil {
+		return nil, fmt.Errorf("failed to get %s secret. %v", monSecret, err)
+	}
+
+	fsid := string(scMon.Data["fsid"])
+	if fsid == "" {
+		return nil, fmt.Errorf("secret %s data fsid is empty", monSecret)
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: monSecret,
+		Kind: "Secret",
+		Data: mustMarshal(map[string]string{
+			"fsid":         fsid,
+			"mon-secret":   "mon-secret",
+			"admin-secret": "admin-secret",
+		})})
+
+	// Service for monitoring endpoints
+	scMonitoring := &v1.Service{}
+	err = s.client.Get(ctx, types.NamespacedName{Name: "rook-ceph-mgr", Namespace: s.namespace}, scMonitoring)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
+	}
+
+	if scMonitoring.Spec.ClusterIP == "" || strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)) == "" {
+		return nil, fmt.Errorf("service rook-ceph-mgr clusterIP or port is empty")
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: "monitoring-endpoint",
+		Kind: "CephCluster",
+		Data: mustMarshal(map[string]string{
+			"MonitoringEndpoint": scMonitoring.Spec.ClusterIP,
+			"MonitoringPort":     strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)),
+		})})
+
+	for _, i := range consumerResource.Status.CephResources {
+		switch i.Kind {
+		case "CephClient":
+			cephClient, err := s.getCephClient(ctx, i.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			cephUserSecret := &v1.Secret{}
+			err = s.client.Get(ctx, types.NamespacedName{Name: cephClient.Status.Info["secretName"], Namespace: s.namespace}, cephUserSecret)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get %s secret. %v", cephClient.Status.Info["secretName"], err)
+			}
+
+			idProp := "userID"
+			keyProp := "userKey"
+			if strings.Contains(i.Name, "-cephfs-") {
+				idProp = "adminID"
+				keyProp = "adminKey"
+			}
+			extR = append(extR, &pb.ExternalResource{
+				Name: cephClient.Status.Info["secretName"],
+				Kind: "Secret",
+				Data: mustMarshal(map[string]string{
+					idProp:  i.Name,
+					keyProp: string(cephUserSecret.Data[i.Name]),
+				}),
+			})
+		case "CephBlockPool":
+			nodeCephClient, err := s.getCephClient(ctx, i.CephClients["node"])
+			if err != nil {
+				return nil, err
+			}
+
+			provisionerCephClient, err := s.getCephClient(ctx, i.CephClients["provisioner"])
+			if err != nil {
+				return nil, err
+			}
+
+			extR = append(extR, &pb.ExternalResource{
+				Name: "ceph-rbd",
+				Kind: "StorageClass",
+				Data: mustMarshal(map[string]string{
+					"clusterID":                 s.namespace,
+					"pool":                      i.Name,
+					"imageFeatures":             "layering",
+					"csi.storage.k8s.io/fstype": "ext4",
+					"imageFormat":               "2",
+					"csi.storage.k8s.io/provisioner-secret-name":            provisionerCephClient.Status.Info["secretName"],
+					"csi.storage.k8s.io/provisioner-secret-namespace":       s.namespace,
```

Comment:
This is incorrect, we cannot assume the namespace on the consumer is the same as on the provider. remove this overwrite.